### PR TITLE
[LETS-14] transaction server connect to page server

### PR DESCRIPTION
**Pull request by radu-vtv**
http://jira.cubrid.org/browse/LETS-14

During boot-up, the transaction server opens a communication_channel to the page server defined in the configuration; failure is logged. 

The page server accepts the incoming channel connection and logs "success".

### CODE REVIEW

File: src/base/server_type.cpp [hunk 1]
```diff
@@ -73,8 +81,35 @@ void init_page_server_hosts ()
...
+}
+
+void connect_to_pageserver (std::string host, int port, const char* db_name)
+{
+  assert (get_server_type () == SERVER_TYPE_TRANSACTION);
+
+  // connect to page server
+  cubcomm::server_channel srv_chn (db_name);
+
+  srv_chn.set_channel_name ("TODO");
```
Comment (ribram) [inline src/base/server_type.cpp:100]: Maybe this:
```suggestion
  srv_chn.set_channel_name ("ATS_PS_comm");
```

File: src/base/server_type.cpp [hunk 2]
```diff
@@ -73,8 +81,35 @@ void init_page_server_hosts ()
...
+  srv_chn.set_debug_dump_data (true); // TODO: is_debug_communication_data_dump_enabled ());
```
Comment (ribram) [inline src/base/server_type.cpp:101]: We discussed to remove debug dump data for now.

File: src/communication/communication_server_channel.hpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+#include "communication_channel.hpp"
+
+#include <string>
+
+namespace cubcomm
+{
+  enum class server_server //server to server commands
+  {
+    CONNECT_TRANSACTION_SERVER, // transaction to page server
+    CONNECT_SLAVE_CONTROL, //slave server to a control channel
```
Comment (ribram) [inline src/communication/communication_server_channel.hpp:36]: Remove channel type specific to HA redesign
```suggestion
    CONNECT_TRANSACTION_SERVER, // transaction to page server
```

File: src/communication/communication_channel.cpp [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+
+#include "communication_channel.hpp"
+
+#include "connection_support.h"
+#include "connection_globals.h"
+
+#include "string_buffer.hpp"
+#include "system_parameter.h"
+#if defined(WINDOWS)
+#include "wintcp.h"
```
Comment (ribram) [inline src/communication/communication_channel.cpp:33]: Without dump data, string_buffer & system_parameter are not required.
```suggestion
#include "connection_support.h"
#include "connection_globals.h"

#if defined(WINDOWS)
#include "wintcp.h"
```

File: src/communication/communication_server_channel.cpp [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+ *
+ */
+
+/*
+ * communication_server_channel.cpp - a communication channel with "connect" function overriden
+ *                                    to match the protocol expected by cub_master
+ */
+
+#include "communication_server_channel.hpp"
+#include "system_parameter.h"  /* er_log_debug param */
```
Comment (ribram) [inline src/communication/communication_server_channel.cpp:26]:
```suggestion
#include "communication_server_channel.hpp"

#include "error_manager.h"
#include "system_parameter.h"  /* er_log_debug param */
```

File: src/connection/connection_defs.h [hunk 6]
```diff
@@ -150,7 +151,8 @@ enum css_server_request
   SERVER_REGISTER_HA_PROCESS = 10,
   SERVER_CHANGE_HA_MODE = 11,
   SERVER_DEREGISTER_HA_PROCESS = 12,
-  SERVER_GET_EOF = 13
+  SERVER_GET_EOF = 13,
+  SERVER_SERVER_CONNECT = 15,
```
Comment (ribram) [inline src/connection/connection_defs.h:155]:
```suggestion
  SERVER_GET_EOF = 13,
  SERVER_SERVER_CONNECT = 14,
```

File: src/connection/connection_defs.h [hunk 7]
```diff
@@ -69,6 +69,7 @@ enum css_command_type
   SERVER_REQUEST = 3,		/* let new server attach */
   UNUSED_REQUEST = 4,		/* unused request - leave it for compatibility */
   SERVER_REQUEST_NEW = 5,	/* new-style server request */
+  CMD_SERVER_SERVER_CONNECT,
```
Comment (ribram) [inline src/connection/connection_defs.h:72]:
```suggestion
  CMD_SERVER_SERVER_CONNECT = 6,
```

**Review by ribram — COMMENTED** [review]
New files license header need updating to new CUBRID license header

File: src/base/server_type.cpp (hunk 1 above)
Comment (radu-vtv) [reply src/base/server_type.cpp]: done

File: src/base/server_type.cpp (hunk 2 above)
Comment (radu-vtv) [reply src/base/server_type.cpp]: done

File: src/connection/server_support.c [hunk 8]
```diff
@@ -2620,7 +2625,37 @@ xacl_reload (THREAD_ENTRY * thread_p)
...
+      assert (false);
+      return;
+    }
+  cubcomm::channel chn;
+  chn.accept (slave_fd);
+  int request = css_get_master_request (slave_fd);	//read an integer to determine connection type
+  switch (STATIC_CAST (cubcomm::server_server, request))
+    {
+    case cubcomm::server_server::CONNECT_TRANSACTION_SERVER:
+      chn.set_channel_name ("TODO");	//cubreplication::REPL_ONLINE_CHANNEL_NAME);
```
Comment (ribram) [inline src/connection/server_support.c:2652]:
```suggestion
      chn.set_channel_name ("ATS_PS_comm");
```

File: msg/fr_FR.utf8/cubrid.msg [hunk 9]
```diff
@@ -1284,7 +1284,7 @@ $ LOADDB
 1214 Reserved error for JSON.
 1215 Plus de données à lire dans le flux %1$s à la position %2$lld, montant %3$zu.
 1216 Identifiant d'objet invalide %1$d lors de la tentative de décompression.
-1217 Reserved error for stream/replication.
+1217 Error in connection with page server.
```
Comment (ribram) [inline msg/fr_FR.utf8/cubrid.msg:1287]:
```suggestion
1217 Reserved error for stream/replication.
```

File: msg/ja_JP.utf8/cubrid.msg [hunk 10]
```diff
@@ -1284,7 +1284,7 @@ $ LOADDB
 1214 Reserved error for JSON.
 1215 No more data to read from stream %1$s at position %2$lld, amount %3$zu.
 1216 Invalid object id %1$d when trying to unpack.
-1217 Reserved error for stream/replication.
+1217 Error connecting with the page server at %s.
```
Comment (ribram) [inline msg/ja_JP.utf8/cubrid.msg:1287]:
```suggestion
1217 Reserved error for stream/replication.
```

Comment (radu-vtv) [reply msg/ja_JP.utf8/cubrid.msg]: done

File: msg/fr_FR.utf8/cubrid.msg (hunk 9 above)
Comment (radu-vtv) [reply msg/fr_FR.utf8/cubrid.msg]: done

File: src/connection/server_support.c (hunk 8 above)
Comment (radu-vtv) [reply src/connection/server_support.c]: done

File: src/connection/connection_defs.h (hunk 7 above)
Comment (radu-vtv) [reply src/connection/connection_defs.h]: done

File: src/connection/connection_defs.h (hunk 6 above)
Comment (radu-vtv) [reply src/connection/connection_defs.h]: done

File: src/communication/communication_server_channel.cpp (hunk 5 above)
Comment (radu-vtv) [reply src/communication/communication_server_channel.cpp]: done

File: src/communication/communication_channel.cpp (hunk 4 above)
Comment (radu-vtv) [reply src/communication/communication_channel.cpp]: done

File: src/communication/communication_server_channel.hpp (hunk 3 above)
Comment (radu-vtv) [reply src/communication/communication_server_channel.hpp]: done

**Review by ribram — APPROVED** [review]

**Pull request merged**